### DISCUSSION TITLE
How do you handle Golang template files?

### DISCUSSION THREAD

**Discussion by dhax** [LazyVim/LazyVim#1775] [Q&A]
Hi there,

having a mostly vanilla LazyVim config with lang.go and formatting.prettier extras enabled, for go html/template files I first need to set the filetype to gotmpl to get gopls attached and then back to html for both go and html syntax highlighting and formatting to work.
I do this with an autocommand if files with pattern *.gohtml,*gotmpl,*.html, etc (there doesn't seem to be a 'standard' file extension?) contain the template string "{{.\\+}}". It seems to work but I am sure there is a better way to achieve this.
Anyone working with go templates, text, html or yaml maybe and got this figured out?

Thanks and have a nice day everyone

**Comment by Moanrisy** [comment]
How did  you match pattern with autocommand?
```

vim.api.nvim_create_autocmd("FileType", {

  pattern = {
    ".*\\.conf",
    ".*.conf",
    "*.conf",
    ".conf",
  },

  command = "setlocal commentstring=#%s",
})
```

I'm using that but .conf still not detected, it still have empty filetype

**Reply by dpetka2001** [reply]
In a `.conf` file do `:set ft?` and see what the message with the filetype is. Use that in the autocmd pattern.

**Reply by Moanrisy** [reply]
I already run that, trying using :verbose set filetype? too

but both only returm empty

`filetype=`

**Reply by dpetka2001** [reply]
When I create an empty file `test.conf` and do `set ft?` the message returns that `filetype=conf`. I don't know what exactly you're doing.

**Reply by Moanrisy** [reply]
How about custom filetype?

test.mytype
or
file.mytype2

is it gonna return filetype=mytype if you run set ft?

**Reply by dpetka2001** [reply]
Well, no. Neovim is only aware of certain filetypes. If you want to dabble with custom filetypes, you should look into `:h vim.filetype.add()`.

**Reply by Moanrisy** [reply]
Already try with `vim.filetype.add()` but got no luck

And I finally got it working using this

```
vim.api.nvim_create_autocmd("BufWinEnter", {
  pattern = "*.conf",
  command = "setlocal filetype=conf | setlocal commentstring=#%s",
})

vim.api.nvim_create_autocmd("BufWinEnter", {
  pattern = "*.mytype2",
  command = "setlocal filetype=mytype2 | setlocal commentstring=#amazingtype2%sendstatement",
})

```

**Reply by dpetka2001** [reply]
Glad you got it working, but I'm sure if you added `vim.filetype.add()` in your `init.lua` file it would have worked just the same. Or create a `filetype.lua` file at the root of your config, where your `init.lua` resides with the following contents
```lua
vim.filetype.add({
  pattern = {
    [".*%.mytype"] = "conf",
  },
})
```
I tried it and it worked. But you can definitely use `BufWinEnter` with the pattern as you specified it as well.

**Comment by dhax** [comment]
currently doing this:
```
vim.api.nvim_create_autocmd({ "BufNewFile", "BufRead" }, {
	pattern = "*.gohtml,*.gotmpl,*.html",
	callback = function()
		if vim.fn.search("{{.\\+}}", "nw") ~= 0 then
			local buf = vim.api.nvim_get_current_buf()
			vim.api.nvim_buf_set_option(buf, "filetype", "gotmpl")
			vim.api.nvim_buf_set_option(buf, "filetype", "html")
		end
	end,
})

```
